### ISSUE TITLE
eslint-disable / eslint-enable still tries to parse content between comments

### ISSUE THREAD

**Issue by whomba** [eslint/eslint#5005]
When an error occurs that you do not care about (such as ESLint thinking a Ruby include should be rendered as JS) the use of:
```
<% extend ActionView::Helpers::JavaScriptHelper %> // eslint-disable-line
```
and 
```    
/* eslint-disable */
  <% extend ActionView::Helpers::JavaScriptHelper %>
/* eslint-enable */
```
This still causes an error to happen, ESLint is still executing this code

Version
v1.10.3

What I did:
* Created a new .eslintrc.json from the eslint --init tool
* Executed: eslint -c .eslintrc.json myFile.js.erb  --ext .js,.es6,.jsx,.erb
* contents of myFile.js.erb:
```
<% extend ActionView::Helpers::JavaScriptHelper %>
```

Output from ESLint:
```
myFile.js.erb
  1:3  error  Parsing error: Unexpected token
```

What I expected to happen:
I would have expected this to pass without an issue as the problematic line has been commented out.

Edit: crucial information for bug reports

**Comment by eslintbot** [comment]
@whomba Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by whomba** [comment]
@eslintbot my apologies, this has been updated

**Comment by platinumazure** [comment]
Yeah, the whole file will be parsed anyway before ESLint comments are taken into account. What you need is a plugin that will preprocess .ejs files which will return only the JavaScript, before parsing and linting.

Do a search on npm to see if there is a plugin for .ejs files.

**Comment by whomba** [comment]
@platinumazure Ah, ok, that makes sense.  I think the documentation should be updated to specify that this only applies to rules to avoid any confusions from people coming from the jshint world.

**Comment by platinumazure** [comment]
@whomba Sounds good- want to write up a PR?

**Comment by btmills** [comment]
Working on this.

**Comment by akashnimare** [comment]
I'm getting the same error on -
```javascript
if (true) {
	console.log("true");
        return;
}
```
Is there any workaround for this? Is there any way I can disable **Parsing error: return outside of function** ?
cc @btmills, @ilyavolodin

**Comment by platinumazure** [comment]
@akashnimare This is a different issue. Take a look at the parser options you can specify in [configuration](http://eslint.org/docs/user-guide/configuring#specifying-parser-options)-- basically you need to avoid `sourceType: module` and possibly add `globalReturn: true`, if you're using the default parser. If that doesn't work, please open a new issue or stop by our [Gitter chat](https://gitter.im/eslint/eslint).

**Comment by akashnimare** [comment]
@platinumazure thanks. I'm using [XO](https://github.com/sindresorhus/xo). My config file is -
```json
  "xo": {
    "esnext": true,
    "overrides": [
      {
        "files": "app*/**/*.js",
        "parserOptions": {
          "ecmaFeatures": {
            "globalReturn": true
            }
            },
            "rules": {
          "max-lines": [
            "warn",
            500
          ],
          "no-warning-comments": 0,
          "capitalized-comments": 0,
          "no-else-return": 0,
          "no-path-concat": 0,
          "no-alert": 0,
          "guard-for-in": 0,
          "prefer-promise-reject-errors": 0,
          "import/no-unresolved": 0,
          "import/no-extraneous-dependencies": 0
        }
      }
    ],
    "ignore": [
      "tests/*.js"
    ],
    "envs": [
      "node",
      "browser",
      "mocha"
    ]
  }
```

**Comment by platinumazure** [comment]
@akashnimare I guess you didn't see my note about opening a new issue for your case since your issue is unrelated to this issue. :smile: If you're using eslint-plugin-import then you are probably using `sourceType: module`, so global return is forbidden in espree for ES6 modules. It's not valid JavaScript.

**Comment by akashnimare** [comment]
My bad, I'll open a new issue. Thanks for explaining :)
